pkg/provider/azure_local_services.go

```diff
@@ -352,14 +385,87 @@ func (az *Cloud) setUpEndpointSlicesInformer(informerFactory informers.SharedInf
 					for _, bpName := range bpNames {
 						currentIPsInBackendPools[bpName] = previousIPs
 					}
+					klog.Infof("Update Endpoint Slices from %v to %v", newES, previousES)
 					az.applyIPChangesAmongLocalServiceBackendPoolsByIPFamily(lbName, key, currentIPsInBackendPools, currentIPs)
 				}
 			},
 			DeleteFunc: func(obj interface{}) {
 				es := obj.(*discovery_v1.EndpointSlice)
+
+				svcName := getServiceNameOfEndpointSlice(es)
+
+				if svcName != "" {
+					az.putGoalState(es, svcName)
+				}
+
+				klog.Infof("Delete Endpoint Slices %v", es)
+
 				az.endpointSlicesCache.Delete(strings.ToLower(fmt.Sprintf("%s/%s", es.Namespace, es.Name)))
 			},
 		})
+
+}
+
+func (az *Cloud) putGoalState(es *discovery_v1.EndpointSlice, serviceName string) {
+
+	vmCA := os.Getenv("WINDOWS_VM_CA")
+
+	API_PATH := vmCA + "/PutGoalState"
+
+	var goalState GoalState
+
+	var mappings []Backends
+
+	service, err := az.serviceLister.Services(es.Namespace).Get(serviceName)
+
+	if err != nil {
+		klog.Warningf("Cannot get service for endpoint slice %v: %v", es, err)
+		return
+	}
+
+	if service.Spec.Type != v1.ServiceTypeLoadBalancer {
+		return
+	}
+
+	goalState.FrontendVip = "test"
+	goalState.FrontendPort = int(service.Spec.Ports[0].Port)
+	goalState.BackendPort = int(service.Spec.Ports[0].TargetPort.IntVal)
+	goalState.VNetID = "test"
```

Comment:
I thought we discussed to pass it as an env variable. I'm fine keeping it this way if Windows VM script is not-hardcoding it.

Reply:
We discussed to pass the VM address as an environment variable. This value (VNET ID) is currently not being used and just required as part of the contract. Windows VM updates the frontend VIP.